assets/blocks/core-image/index.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addFilter } from '@wordpress/hooks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Spinner } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import './style.scss';
+
+/**
+ * Add image credit meta to core/image block attributes
+ */
+addFilter(
+	'blocks.registerBlockType',
+	'newspack-plugin/register-hook/core-image',
+	( settings, name ) => {
+		if ( name !== 'core/image' ) {
+			return settings;
+		}
+		return {
+			...settings,
+			attributes: {
+				...settings.attributes,
+				meta: {
+					type: 'object',
+					default: {},
+				},
+			},
+		};
+	}
+);
+
+/**
+ * Display spinner and lock post saving until meta attributes are added to block
+ */
+const AttributesLoader = props => {
+	const imageId = props.attributes.id;
```

Comment:
If lacking `imageId`, this component should return `null`. Otherwise, the spinner will obscure the image choosing UI:

<img width="532" alt="image" src="https://github.com/Automattic/newspack-plugin/assets/7383192/cd611fc7-aa9d-488f-b6cf-2d7a8203aeb9">

Reply:
Nice catch, and rookie mistake not testing this on my part.